EventHubs/tests/test_eventhub_metrics.py

```diff
@@ -1,48 +1,151 @@
+import sys
 import unittest
-import datetime
+from io import StringIO
+from datetime import timedelta
 from baseeventhubtest import BaseEventHubTest
-
-
+from azure.monitor.query import LogsQueryClient, LogsQueryStatus
 class TestEventHubMetrics(BaseEventHubTest):
 
     def setUp(self):
-        super(TestEventHubMetrics, self).setUp()
-        self.RESOURCE_GROUP_NAME = "TestEventHubMetrics-%s" % (
-            datetime.datetime.now().strftime("%d-%m-%y-%H-%M-%S"))
-        self.function_name_prefix = "EventHubs_Metrics"
-        self.STORAGE_ACCOUNT_NAME = "sumometlogs"
-        self.template_name = 'azuredeploy_metrics.json'
-        self.log_table_name = "AzureWebJobsHostLogs%d%02d" % (
-            datetime.datetime.now().year, datetime.datetime.now().month)
+        self.RESOURCE_GROUP_NAME = "azure_metric_unittest"
+        self.namespace_location = "eastus"
+        # self.function_name_prefix = "EventHubs_Metrics"
+        # self.STORAGE_ACCOUNT_NAME = "sumometlogs"
+        self.template_name = "azuredeploy_metrics.json"
         self.event_hub_namespace_prefix = "SumoMetricsNamespace"
-        self.eventhub_name = 'insights-metrics-pt1m'
-
+        self.eventhub_name = "insights-metrics-pt1m"
+        super(TestEventHubMetrics, self).setUp()
+        
     def test_pipeline(self):
-
         self.create_resource_group()
-        self.deploy_template()
+        # self.deploy_template()
         print("Testing Stack Creation")
-        self.assertTrue(self.resource_group_exists(self.RESOURCE_GROUP_NAME))
-        self.table_service = self.get_table_service()
-        self.insert_mock_logs_in_EventHub('metrics_fixtures.json')
-        self.check_error_logs()
+        self.assertTrue(self.resource_group_exists(self.RESOURCE_GROUP_NAME)) 
+        # total resource 8 or 9
+        self.insert_mock_metrics_in_EventHub('metrics_fixtures.json')
+        self.check_success_log()
+        self.check_error_log()
+        self.check_warning_log()
+    
+    def check_success_log(self):
+        # Save the original stdout for later comparison
+        original_stdout = sys.stdout
+        sys.stdout = StringIO()
+
+        try:
+            client = LogsQueryClient(self.azure_credential)
+            
+            query = '''union
```

Comment:
have a common function which will take in the query and returns output currently this query_workspace logic is repeated at 3 places.

Reply:
Updated